### PR TITLE
Improve documentation of API kinds

### PR DESCRIPTION
**Pull request by nafur**
This PR improves the documentation of the `api::Kind` enum. Note that the docs for many of the enum values should still be improved. This PR merely makes sure that everything that is already there is actually output (`/*` vs `/**`) and properly rendered (missing spacing between lists, some formulas, etc).

### CODE REVIEW

File: docs/cpp/kind.rst [hunk 1]
```diff
@@ -1,10 +1,16 @@
 Kind
 ====
 
+Every :cpp:class:`Term <cvc5::api::Term>` has a kind which represents its type, for example whether it is an equality (:cpp:enumerator:`EQUAL <cvc5::api::Kind::EQUAL>`), a conjunction (:cpp:enumerator:`AND <cvc5::api::Kind::AND>`), or a bitvector addition (:cpp:enumerator:`BITVECTOR_PLUS <cvc5::api::Kind::BITVECTOR_PLUS>`).
+
+Note that the api type :cpp:enum:`cvc5::api::Kind` roughly corresponds to :cpp:enum:`cvc5::Kind`, but is a different type and has a few subtle differences.
```
Comment (aniemetz) [inline docs/cpp/kind.rst:6]:
```suggestion
Note that the API type :cpp:enum:`cvc5::api::Kind` roughly corresponds to :cpp:enum:`cvc5::Kind`, but is a different type and has a few subtle differences.
```

Comment (aniemetz) [reply docs/cpp/kind.rst:6]: I'm not sure if we should even mention that. `cvc5::Kind` should not be accessible from outside at all.

File: src/api/cpp/cvc5_kind.h [hunk 2]
```diff
@@ -78,187 +82,227 @@ enum CVC4_EXPORT Kind : int32_t
...
    * Permitted in bindings and in the lambda and quantifier bodies only.
+   * 
    * Parameters:
-   *   See mkVar().
+   *   - See @ref cvc5::api::Solver::mkVar() "mkVar()".
+   * 
    * Create with:
-   *   mkVar(const std::string& symbol, Sort sort)
-   *   mkVar(Sort sort)
+   *   - `cvc5::api::Solver::mkVar(Sort sort, const std::string& symbol)`
```
Comment (aniemetz) [inline src/api/cpp/cvc5_kind.h:129]: Why do we need prefix `cvc5::api` here but not above, for `mkConst` or `mkTerm`?

File: src/api/cpp/cvc5_kind.h [hunk 3]
```diff
@@ -2845,7 +3405,7 @@ enum CVC4_EXPORT Kind : int32_t
 #endif
 
   /* ----------------------------------------------------------------------- */
-  /* marks the upper-bound of this enumeration */
+  /** marks the upper-bound of this enumeration */
```
Comment (aniemetz) [inline src/api/cpp/cvc5_kind.h:3408]:
```suggestion
  /** Marks the upper-bound of this enumeration. */
```

**Review by aniemetz — COMMENTED** [review]
Again, 80 chars for .rst.

File: src/api/cpp/cvc5_kind.h (hunk 2 above)
Comment (nafur) [reply src/api/cpp/cvc5_kind.h]: We don't need it.

File: docs/cpp/kind.rst [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
 ====
 
+Every :cpp:class:`Term <cvc5::api::Term>` has a kind which represents its type,
+for example whether it is an equality (:cpp:enumerator:`EQUAL
+<cvc5::api::Kind::EQUAL>`), a conjunction (:cpp:enumerator:`AND
+<cvc5::api::Kind::AND>`), or a bitvector addition
+(:cpp:enumerator:`BITVECTOR_PLUS <cvc5::api::Kind::BITVECTOR_PLUS>`).
+
+Note that the API type :cpp:enum:`cvc5::api::Kind` roughly corresponds to
+:cpp:enum:`cvc5::Kind`, but is a different type and has a few subtle
```
Comment (aniemetz) [inline docs/cpp/kind.rst:11]:
```suggestion
:cpp:enum:`cvc5::Kind`, but is a different type. It hides internal kinds that
should not be exported to the API, and maps all kinds that we want to export
to its corresponding internal kinds.
```

File: docs/cpp/kind.rst [hunk 5]
```diff
@@ -1,10 +1,22 @@
...
+differences.
```
Comment (aniemetz) [inline docs/cpp/kind.rst:12]:
```suggestion
```

**Review by aniemetz — APPROVED** [review]

**Pull request merged**